input.go

```diff
@@ -23,6 +23,26 @@ func (m Keymap) Clone() Keymap {
 	return cloned
 }
 
+// MergeKeymaps merges a list of maps into one.
+// Given maps are not modified.
+// Resulting map contains no references to given maps.
+func MergeKeymaps(maps ...Keymap) Keymap {
+	merged := make(Keymap)
+	for _, m := range maps {
+		for k, list := range m {
+			existing, ok := merged[k]
+			if !ok {
```

Comment:
Is this separate branch act as an optimization?
I wonder what's the reason to avoid an easier 1-liner like
```go
for k, list := range m {
  merged[k] = append(merged[k], list...)
}
```

As a nitpick: `k` should probably be named `a` (`Action`); otherwise, it looks like we're mapping keys to keys, which could be confusing.

Reply:
I believe I have addressed all of those suggestions in two commits. Thank you!